src/main/java/services/impl/InstallServiceImpl.java

```diff
@@ -23,23 +25,28 @@
     @Override
     public CommCareConfigEngine configureApplication(String reference, final String username, final String dbPath) {
         log.info("Configuring application with reference " + reference + " and dbPath: " + dbPath + ".");
-
-        final String trimmedUsername = StringUtils.substringBefore(username, "@");
-
-        File dbFolder = new File(dbPath);
-        dbFolder.delete();
-        dbFolder.mkdirs();
-
-        CommCareConfigEngine.setStorageFactory(new IStorageIndexedFactory() {
-            @Override
-            public IStorageUtilityIndexed newStorage(String name, Class type) {
-                return new SqliteIndexedStorageUtility(type, name, trimmedUsername, dbPath);
-            }
-        });
-
-        CommCareConfigEngine engine = new FormplayerConfigEngine(PrototypeManager.getDefault());
-        engine.initFromArchive(reference);
-        engine.initEnvironment();
-        return engine;
+        try {
+            final String trimmedUsername = StringUtils.substringBefore(username, "@");
+
+            File dbFolder = new File(dbPath);
+            SqlSandboxUtils.deleteDatabaseFolder(dbPath);
+            dbFolder.mkdirs();
+
+            CommCareConfigEngine.setStorageFactory(new IStorageIndexedFactory() {
+                @Override
+                public IStorageUtilityIndexed newStorage(String name, Class type) {
+                    return new SqliteIndexedStorageUtility(type, name, trimmedUsername, dbPath);
+                }
+            });
+
+            CommCareConfigEngine engine = new FormplayerConfigEngine(PrototypeManager.getDefault());
+            engine.initFromArchive(reference);
+            engine.initEnvironment();
+            return engine;
+        } catch (Exception e) {
+            log.error("Got exception " + e + " while installing reference " + reference + " at path " + dbPath);
+            SqlSandboxUtils.deleteDatabaseFolder(dbPath);
+            throw e;
```

Comment:
do we need to convert this to a runtime exception so that it's handled correctly?